src/integrationTest/java/uk/gov/hmcts/reform/managecase/repository/DefinitionStoreRepositoryIT.java

```diff
@@ -0,0 +1,117 @@
+        CacheManager cacheManager() {
+            return new ConcurrentMapCacheManager("challengeQuestions", "caseRoles");
+        }
+
+        @Bean
+        DefinitionStoreRepository definitionStoreRepository() {
+            initMocks(this);
+            FieldType fieldType = FieldType.builder()
+                .regularExpression("regular expression")
+                .max(null)
+                .min(null)
+                .id("Number")
+                .type("Number")
+                .build();
+            ChallengeQuestion challengeQuestion = ChallengeQuestion.builder()
+                .caseTypeId(CASE_TYPE_ID)
+                .challengeQuestionId("NoC")
+                .questionText("questionText")
+                .answerFieldType(fieldType)
+                .answerField(ANSWER_FIELD_APPLICANT)
+                .questionId("QuestionId1")
+                .order(1).build();
+            ChallengeQuestionsResult challengeQuestionsResult = new ChallengeQuestionsResult(
+                Arrays.asList(challengeQuestion));
+            when(definitionStoreApiClient.challengeQuestions(CASE_TYPE_ID, CASE_ID))
+                .thenReturn(challengeQuestionsResult);
+
+            List<CaseRole> caseRoleList = Arrays.asList(
+                CaseRole.builder().id("APPLICANT").name("Applicant").build());
+
+            when(definitionStoreApiClient.caseRoles(USER_ID, JURISDICTION, CASE_TYPE_ID)).thenReturn(caseRoleList);
+
+            CaseTypeDefinitionVersion caseTypeDefinitionVersion = new CaseTypeDefinitionVersion();
+            caseTypeDefinitionVersion.setVersion(123);
+            when(definitionStoreApiClient.getLatestVersion(CASE_TYPE_ID)).thenReturn(caseTypeDefinitionVersion);
+
+            return new DefaultDefinitionStoreRepository(definitionStoreApiClient);
+        }
+    }
+
+    @Autowired
+    private DefinitionStoreRepository repository;
+    @Autowired
+    private CacheManager cacheManager;
+
+    private static final String ANSWER_FIELD_APPLICANT = "${applicant.individual.fullname}|${applicant.company.name}|"
+        + "${applicant.soletrader.name}|${OrganisationPolicy1"
+        + ".Organisation.OrganisationID}:Applicant";
+    private static final String CASE_ID = "12345678";
+    private static final String CASE_TYPE_ID = "TEST_CASE_TYPE";
+    private static final String ZERO = "0";
+    private static final String JURISDICTION = ZERO;
+    private static final String USER_ID = ZERO;
+
+    @Test
+    public void verifyChallengeQuestionsCache() {
+        repository.challengeQuestions(CASE_TYPE_ID, CASE_ID);
+        assertNotNull(cacheManager.getCache("challengeQuestions").getNativeCache());
+
+        cacheManager.getCache("challengeQuestions").clear();
+        assertEquals(cacheManager.getCache("challengeQuestions").getNativeCache().toString(), "{}");
+
+        repository.challengeQuestions(CASE_TYPE_ID, CASE_ID);
+        assertNotNull(cacheManager.getCache("challengeQuestions").getNativeCache());
```

Comment:
could you assert number of calls it's making to definitionStoreApiClient for each cache hit/miss